ViAn/Video/overlay.cpp

```diff
@@ -94,3 +100,40 @@ void overlay::set_overlay_tool(SHAPES s) {
 void overlay::set_overlay_colour(QColor col) {
     colour = col;
 }
+
+/**
+ * @brief overlay::mouse_pressed
+ * Starts drawing on the overlay when the mouse is
+ * pressed, if the overlay is visible.
+ * @param pos coordinates
+ */
+void overlay::mouse_pressed(QPoint pos) {
+    if (show_overlay) {
+        draw_start = pos;
+        draw_end = pos;
+    }
+}
+
+/**
+ * @brief overlay::mouse_pressed
+ * Ends drawing on the overlay when the mouse is
+ * released, if the overlay is visible.
+ * @param pos coordinates
+ */
+void overlay::mouse_released(QPoint pos) {
```

Comment:
These will not be identical in future, but I've extracted the code to make it look nicer.